### DISCUSSION TITLE
How to use "Whisper" to detect whether there is a human voice in an audio segment？

### DISCUSSION THREAD

**Discussion by TroyMitchell911** [openai/whisper#1907] [Q&A]
I am developing a voice assistant that implements the function of stopping recording and saving audio files when no one is speaking, based on volume. If this audio file is directly used for Whisper recognition, it is possible to mistakenly recognize it as another language

**Comment by EtienneAb3d** [comment]
@JiangLin998 
To detect whether there is a human voice in an audio segment, I would:

1. use Demucs to extract voice from any other kind of noise/music
2. use SileroVAD and FFmpeg to cut silence parts and possible remaining noise parts
3. use FFmpeg to check the duration of the resulting file: if zero, there is no voice in

See WhisperHallu to get a code example for each step:
https://github.com/EtienneAb3d/WhisperHallu
